merkletree/proof.go

```diff
@@ -59,7 +75,9 @@ func (ap *AuthenticationPath) authPathHash() []byte {
 	return hash
 }
 
-// Verify should be called after the vrf index is verified successfully
+// Verify recomputes the tree's root node from the authentication path,
+// and compares it to the tree hash in the corresponding signed tree root.
```

Comment:
This sentence could be a bit confusing since this function doesn't use any STRs. Maybe mention the `treeHash` parameter explicitly and mention that it has to come from an STR?